### PR TITLE
HDDS-6120. Compute block checksum using chunk checksums

### PR DESCRIPTION
**Pull request by jojochuang**
This PR depends on HDDS-6086.

## What changes were proposed in this pull request?

Compute block checksum using checksumData from chunkInfo.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6120

## How was this patch tested?

Added/Modified unit and end-to-end tests

### CODE REVIEW

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/ReplicatedBlockChecksumComputer.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import org.apache.hadoop.io.MD5Hash;
+import org.apache.ratis.thirdparty.com.google.protobuf.ByteString;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+class ReplicatedBlockChecksumComputer extends
```
Comment (rakeshadr) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/ReplicatedBlockChecksumComputer.java:30]: Can we move all the checksum related all the newly added classes to a separate package "`org.apache.hadoop.ozone.client.checksum`"

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/ReplicatedBlockChecksumComputer.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    this.chunkInfoList = chunkInfoList;
+  }
+
+  @Override
+  void compute() throws IOException {
+    computeMd5Crc();
+  }
+
+  // compute the block checksum, which is the md5 of chunk checksums
+  private void computeMd5Crc() throws IOException {
```
Comment (rakeshadr) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/ReplicatedBlockChecksumComputer.java:49]: I think, now can add an end-to-end integration test to verify the checksum. Please add few cases.

**Review by rakeshadr — COMMENTED** [review]
Thanks @jojochuang for the patch. Added few comments

**Comment by jojochuang** [comment]
Rebased the PR. Still need to add the tests suggested by Rakesh.

**Review by rakeshadr — APPROVED** [review]
+1 LGTM. Thanks @jojochuang for the contribution.

**Comment by jojochuang** [comment]
Thanks @rakeshadr for the review!

**Pull request merged**